### PR TITLE
Fix statsd metrics not sending when using daemon mode

### LINKED ISSUE #13741: Airflow 2.0 does not send metrics to statsD when Scheduler is run with Daemon mode
**Apache Airflow version**:
2.0.0

**Environment**:

- **OS** (e.g. from /etc/os-release): Ubuntu 20.04 LTS
- **Python version**: 3.8
- **Kernel** (e.g. `uname -a`): x86_64 x86_64 x86_64 GNU/Linux 5.4.0-58-generic #64-Ubuntu
- **Install tools**: pip

**What happened**:

Airflow 2.0 does not send metrics to statsD.

I configure Airflow with official documentation (https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/metrics.html) and by this article https://dstan.medium.com/run-airflow-statsd-grafana-locally-16b372c86524 (but I used port 8125). 

I turned on statsD:

```ini
statsd_on = True
statsd_host = localhost
statsd_port = 8125
statsd_prefix = airflow
```

But I do not see airflow metrics at http://localhost:9102/metrics (statsD metrics endpoint).

---

P.S. I noticed this error just using Airflow 2.0. In version 1.10.13 everything is ok in the same environment.

Thank you for advance.

### PR DESCRIPTION
**Pull request by junnplus**
closes: https://github.com/apache/airflow/issues/13741

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/stats.py [hunk 1]
```diff
@@ -348,25 +348,28 @@ def timer(self, stat=None, *args, tags=None, **kwargs):
...
 class _Stats(type):
+    factory = None
     instance: Optional[StatsLogger] = None
 
     def __getattr__(cls, name):
+        if not cls.instance:
+            try:
+                cls.instance = cls.factory()
+            except (socket.gaierror, ImportError) as e:
+                log.error("Could not configure StatsClient: %s, using DummyStatsLogger instead.", e)
```
Comment (ashb) [inline airflow/stats.py:359]: If a stats backend _isn't_ configured, isn't factory going to be None, and so we'd die with a TypeError?

Comment (ashb) [reply airflow/stats.py:359]: Oh no, cos this is a metaclass.

**Review by ashb — APPROVED** [review]

**Comment by saimon46** [comment]
Hello we still see this problem in version 2.2.3 of Airflow. Can you please recheck at it?

**Comment by potiuk** [comment]
> Hello we still see this problem in version 2.2.3 of Airflow. Can you please recheck at it?

This issue is closed. I tihnk this is another problem. Can you please create a new issue with all the logs and information on how your configuration is? It's likely configuration on your side, so opening a [discussion](https://github.com/apache/airflow/discussions) first, before it will be determined as Airflow issue is preferred (also a bit esier than creating an issue).

**Pull request merged**